aicsimageio/readers/czi_reader.py

```diff
@@ -0,0 +1,795 @@
+                    scene=self.current_scene_index,
+                    retrieve_dims=blocked_dim_order,
+                    retrieve_indices=retrieve_indices,
+                ),
+                shape=chunk_shape,
+                dtype=pixel_type,
+            )
+
+        # Convert the numpy array of lazy readers into a dask array
+        image_data = da.block(lazy_arrays.tolist())
+
+        # Because we have set certain dimensions to be chunked and others not
+        # we will need to transpose back to original dimension ordering
+        # Example, if the original dimension ordering was "TZYX" and we
+        # chunked by "T", "Y", and "X"
+        # we created an array with dimensions ordering "ZTYX"
+        transpose_indices = []
+        for i, d in enumerate(valid_dims):  # selected_scene_dims):
+            new_index = blocked_dim_order.index(d)
+            if new_index != i:
+                transpose_indices.append(new_index)
+            else:
+                transpose_indices.append(i)
+
+        # Only run if the transpose is actually required
+        image_data = da.transpose(image_data, tuple(transpose_indices))
+
+        return image_data
+
+    @staticmethod
+    def _get_coords_and_physical_px_sizes(
+        xml: ET.Element, scene_index: int, image_short_info: Dict[str, Any]
+    ) -> Tuple[Dict[str, Any], types.PhysicalPixelSizes]:
+        # Create coord dict
+        coords = {}
+
+        # Get all images
+        img_sets = xml.findall(".//Image/Dimensions/Channels")
+
+        if len(img_sets) != 0:
+            # Select the current scene
+            img = img_sets[scene_index]
+
+            # Construct channel list
+            scene_channel_list = []
+            channels = img.findall("./Channel")
+            for i, channel in enumerate(channels):
+                channel_name = channel.attrib["Name"]
+                channel_id = channel.attrib["Id"]
+                channel_ce = channel.find("./ContrastMethod")
+                channel_contrast = channel_ce.text if channel_ce is not None else ""
+                scene_channel_list.append(
+                    (f"{channel_id}" f"--{channel_name}" f"--{channel_contrast}")
+                )
+
+            # Attach channel names to coords
+            coords[DimensionNames.Channel] = scene_channel_list
+
+        # Unpack short info scales
+        list_xs = xml.findall(".//Distance[@Id='X']")
+        list_ys = xml.findall(".//Distance[@Id='Y']")
+        list_zs = xml.findall(".//Distance[@Id='Z']")
+
+        scale_xe = list_xs[0].find("./Value")
+        scale_ye = list_ys[0].find("./Value")
+        scale_ze = None if len(list_zs) == 0 else list_zs[0].find("./Value")
+        scale_x = float(str(scale_xe.text)) if scale_xe is not None else 1.0
+        scale_y = float(str(scale_ye.text)) if scale_ye is not None else 1.0
+        scale_z = float(str(scale_ze.text)) if scale_ze is not None else 1.0
+        scale_t = None
```

Comment:
Uhhhhh shouldn't these be per scene? Or is it really consistent scaling for all scenes?

Reply:
Bump.